include/zephyr/drivers/flash/andes_flash_xip_api_ex.h

```diff
@@ -4,70 +4,152 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+/**
+ * @file
+ * @brief Header file for Andes QSPI-NOR XIP flash extended operations.
+ * @ingroup andes_flash_xip_ex_op
+ */
+
 #ifndef __ZEPHYR_INCLUDE_DRIVERS_ANDES_FLASH_XIP_API_EX_H__
 #define __ZEPHYR_INCLUDE_DRIVERS_ANDES_FLASH_XIP_API_EX_H__
 
+/**
+ * @brief Extended operations for Andes QSPI-NOR XIP flash.
+ * @defgroup andes_flash_xip_ex_op Andes QSPI-NOR XIP
+ * @ingroup flash_ex_op
+ * @{
+ */
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 #include <zephyr/drivers/flash.h>
 
+/**
+ * @brief Enumeration for Andes flash extended operations.
+ */
 enum flash_andes_xip_ex_ops {
+	/**
+	 * @brief Get the three status registers (SR1, SR2, SR3).
+	 *
+	 * This operation reads the contents of the three status registers from the flash device.
+	 *
+	 * @param out Pointer to a @ref andes_xip_ex_ops_get_out structure to store the register
+	 *            values.
+	 */
 	FLASH_ANDES_XIP_EX_OP_GET_STATUS_REGS = FLASH_EX_OP_VENDOR_BASE,
+	/**
+	 * @brief Set the three status registers (SR1, SR2, SR3).
+	 *
+	 * This operation writes new values to the status registers, applying a mask to modify only
+	 * specific bits.
+	 * This operation is protected by a software lock that can be controlled with
+	 * @ref FLASH_ANDES_XIP_EX_OP_LOCK.
+	 *
+	 * @param in Pointer to a @ref andes_xip_ex_ops_set_in structure containing the values and
+	 *           masks to write.
```

Comment:
note to self, using `@param` this way is a bit of a Doxygen hack since there aren't actual parameters being documented for this enumerator, but given this gives a really nice output IMO, let's stick to it :)

<img width="1100" height="466" alt="image" src="https://github.com/user-attachments/assets/d656cdef-a999-47ee-8d22-ca44ec13ad84" />

Reply:
yeah it's a bit confusing, surprised the system does not print a warning about it

Reply:
@fabiobaltieri Yeah was surprised too -- it's prob enough of a hack that it's worth cleaning up ; will do so as a follow-up if folks are ok with that